internal/pkg/cgroups/manager_linux_test.go

```diff
@@ -134,31 +135,49 @@ func ensureContainsInt(t *testing.T, path string, wantInt int64) {
 	t.Fatalf("%s did not contain expected value %d", path, wantInt)
 }
 
-// ensureState asserts that a process pid has the required state
-func ensureState(t *testing.T, pid int, wantStates string) {
+// ensureStateBecomes asserts that a process pid has any of the wanted
+// states, or reaches one of these states in a 5 second window.
+func ensureStateBecomes(t *testing.T, pid int, wantStates string) {
+	const retries = 5
+	const delay = time.Second
+
 	file, err := os.Open(fmt.Sprintf("/proc/%d/status", pid))
 	if err != nil {
 		t.Error(err)
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
 	procState := ""
 
-	for scanner.Scan() {
-		// State:	R (running)
-		if strings.HasPrefix(scanner.Text(), "State:\t") {
-			f := strings.Fields(scanner.Text())
-			if len(f) < 2 {
-				t.Errorf("Could not check process state - not enough fields: %s", scanner.Text())
+	for r := 0; r <= retries; r++ {
+		if r > 0 {
+			t.Logf("Process %d has state %q, need %q - retrying %d/%d", pid, procState, wantStates, r, retries)
+			time.Sleep(delay)
+		}
+
+		if _, err := file.Seek(0, 0); err != nil {
```

Comment:
Nit: clearer if you use a `const` for the whence:

```suggestion
		if _, err := file.Seek(0, io.SeekStart); err != nil {
```